proto/frontend/src/pre_mc_mgr.cpp

```diff
@@ -42,6 +45,17 @@ using Code = ::google::rpc::Code;
 using Status = PreMcMgr::Status;
 using GroupEntry = PreMcMgr::GroupEntry;
 
+std::tuple<p4v1::Replica::PortKindCase, pi_mc_port_t, size_t> ReplicaPortAsTuple(
+    const ReplicaPort &port) {
+  return std::make_tuple(port.port_kind, port.port_id, port.num_bytes);
+}
+bool operator==(const ReplicaPort &x, const ReplicaPort &y) {
+  return ReplicaPortAsTuple(x) == ReplicaPortAsTuple(y);
+}
+bool operator<(const ReplicaPort &x, const ReplicaPort &y) {
+  return ReplicaPortAsTuple(x) < ReplicaPortAsTuple(y);
+}
```

Comment:
not sure I see the value of building a tuple here as opposed to comparing individual fields?

Reply:
My motivation was simplicity/readability.
The `<` operator becomes a bit convoluted:
```
bool operator<(const ReplicaPort &x, const ReplicaPort &y) {
  if (x.port_kind < y.port_kind) return true;
  if (y.port_kind < x.port_kind) return false;
  if (x.port_id < y.port_id) return true;
  if (y.port_id < x.port_id) return false;
  return x.num_bytes < y.num_bytes;
}
```
Happy to change it if you prefer the above.